### DISCUSSION TITLE
Problem while deploying backend on Render

### DISCUSSION THREAD

**Discussion by pankajcs120** [community/community#109765] [Programming Help]
### Select Topic Area

Question

### Feature Area

Issues

### Body

Hello Everyone,
I am deploying my backend on render from github repository.  It deploys successfully but when i click on the link given by render. It shows Cannot Get / . Can someone help me on this. Thanks.
My Github Repository :- https://github.com/pankajcs120/OTSBack.git

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @pankajcs120 ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by Ahmad105010** [comment]
It seems like the issue you're encountering is related to routing or configuration in your backend application. The "Cannot Get /" error typically indicates that the server is unable to find a route for the root URL ("/").

Here are a few steps you can take to troubleshoot and potentially resolve the issue:

Check Your Server Configuration: Ensure that your backend server is correctly configured to handle requests for the root URL ("/"). This might involve configuring routes or ensuring that your server's default behavior is set up correctly.

Verify Your Deployment Settings: Double-check the settings for your deployment on Render to ensure that the correct entry point for your application is specified. This could be an incorrect start command or a misconfigured route.

Review Your Code: Look through your backend codebase, specifically any routing configuration or entry point files (e.g., app.js, server.js, etc.), to ensure that the root URL ("/") is properly handled.

Inspect Logs: Check the logs provided by Render or your backend server to see if there are any error messages or clues as to why the server is unable to handle requests for the root URL.

Test Locally: Try running your backend application locally to see if you encounter the same issue. This can help isolate whether the problem is specific to the deployment environment or if it exists in your code.

Without direct access to your deployment settings and codebase, it's challenging to pinpoint the exact cause of the issue. However, going through these steps should help you identify and resolve the problem. If you need further assistance, feel free to provide more details or share relevant code snippets, and I'd be happy to help further!
applicationswriting.com

**Comment by DanielWren48** [comment]
Hi @pankajcs120,
I was looking at your server.js file and your only starter route seems to be
```
app.get('/api', (req, res) => {
  res.send('Hello, World!');
});
```
and maybe render does not pick up on that, have you tried adding a root path like
```
app.get('/', (req, res) => {
  res.send('Hello, World!');
});
```

Also have you redeployed your server to Vercel by now, bc your project shows a link to a vercel?
